src/server/handlers/config.rs

```diff
@@ -36,9 +32,9 @@ impl JsonRPCRequestHandler for ReloadConfigHandler {
     async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
         log::info!("received request to reload config: {request:?}");
 
-        let path = request
-            .path
-            .unwrap_or_else(|| self.default_config_path.clone());
-        self.config.reload(path).await
+        if request.path.is_some() {
+            self.config.set_path(request.path.unwrap());
```

Comment:
Just to double-check, when we start the daemon if no config path is specified the default one is used to construct the `ReloadConfig` right?

Reply:
It uses the one already present in `ReloadableConfig`.